CONTRIBUTING.md

```diff
@@ -2,48 +2,79 @@
-[Contributing Code](#contributing-code)
-  [Licensing And Legal](#licensing-and-legal)
-  [Style Guide](#style-guide)
-  [Pull Requests](#pull-requests)
+1. [Reporting Bugs](#reporting-bugs)
+    1. [Test a fix](#test-a-fix)
+    2. [Writing a Clear Bug Report](#writing-a-clear-bug-report)
+2. [Suggesting Features and/or Enhancements](#suggesting-features-and-enhancements)
+3. [Contributing Code](#contributing-code)
+    1. [Licensing And Legal](#licensing-and-legal)
+    2. [Style Guide](#style-guide)
+    3. [Pull Requests](#pull-requests)
+    5. [Writing content for the changelog](#writing-content-for-the-changelog)
 
 ## Reporting Bugs
 
 We track bugs on the repository's [GitHub issue tracker](https://github.com/highcharts/highcharts/issues).
 
-Before submitting a new bug report, please try to search existing (open and closed) issues to 
-see if the issue is already reported. If you find an existing open issue, please
-add any additional information you might have on the bug to the existing ticket.
+Before submitting a [new bug report](https://github.com/highcharts/highcharts/issues/new/choose),
+please try to search existing (open and closed) issues to see if the issue is
+already reported. If you find an existing open issue, please add any additional
+information you might have on the bug to the existing ticket.
+
+We use GitHub Issues as our official bug tracker. We strive to keep this a
+clean, maintainable and searchable record of our open and closed bugs, therefore
+we kindly ask you to obey some rules before reporting an issue:
+
+1. Make sure the report is accompanied by a reproducible demo. The ideal demo is
+created by forking [our standard jsFiddle](http://jsfiddle.net/highcharts/LLExL/),
+adding your own code and stripping it down to an absolute minimum needed to
+demonstrate the bug.
+2. Always add information on what browser it applies to, and other information
+needed for us to debug.
+3. It may be that the bug is already fixed. See [Test a fix](#test-a-fix).
 
 If you find a closed issue describing your problem, please open a new issue and 
-link to the closed one in your issue description (use `#<ticket number>`, e.g. `#1234`).
+link to the closed one in your issue description (use `#<ticket number>`, e.g.
+`#12345`).
 
 Please also follow the directions in our [ticket submission template](https://github.com/highcharts/highcharts/blob/master/ISSUE_TEMPLATE.md).
 
+### Test a fix
+
+When an issue is resolved, we commit a fix and mark the issue closed. This
+doesn't mean that a new release is available with the fix applied, but that it
+is fixed in the development code and will be added to the next stable release.
+Stable versions are typically released every 1-3 months. To try out the fix
+immediately, you can run http://github.highcharts.com/highcharts.js or
+http://github.highcharts.com/highstock.js from any website, but do not use these
+URLs in production.
```

Comment:
Perhaps we should advertise the nightly here instead/as well?

[jsfiddle for convenience](https://jsfiddle.net/goransle/58nutgqz/)